morphman/automated_landmarking/automated_landmarking_tools.py

```diff
@@ -0,0 +1,245 @@
+##   Copyright (c) Aslak W. Bergersen, Henrik A. Kjeldsberg. All rights reserved.
+##   See LICENSE file for details.
+
+##      This software is distributed WITHOUT ANY WARRANTY; without even
+##      the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR
+##      PURPOSE.  See the above copyright notices for more information.
+
+from scipy.ndimage.filters import gaussian_filter
+from scipy.signal import argrelextrema
+
+# Local import
+from morphman.common import *
+
+
+def get_maximum_coronal_coordinate(coronal_coordinate, length):
+    value_index = coronal_coordinate[argrelextrema(coronal_coordinate, np.less_equal)[0]].min()
+    max_coronal_coordinate_id = np.array(coronal_coordinate.tolist().index(value_index))
+    search_tolerance = 30
+
+    if abs(length[max_coronal_coordinate_id] - length[-1]) > search_tolerance:
+        print("-- Sanity check failed, checking for maximums")
+
+        value_index = coronal_coordinate[argrelextrema(coronal_coordinate, np.greater_equal)[0]].max()
+        max_coronal_coordinate_id = np.array(coronal_coordinate.tolist().index(value_index))
+        if abs(length[max_coronal_coordinate_id] - length[-1]) > search_tolerance:
+            print("-- Sanity check failed, no anterior bend in model. Exiting.")
+            sys.exit(1)
+    return max_coronal_coordinate_id
+
+
+def get_centerline_coordinates(line, length):
+    x = np.zeros(length.shape[0])
+    y = np.zeros(length.shape[0])
+    z = np.zeros(length.shape[0])
+    for i in range(z.shape[0]):
+        x[i] = line.GetPoints().GetPoint(i)[0]
+        y[i] = line.GetPoints().GetPoint(i)[1]
+        z[i] = line.GetPoints().GetPoint(i)[2]
```

Comment:
These six lines could be replaced with:

```python
from vtk.numpy_interface import dataset_adapter as dsa
x, y, z = dsa.WrapDataObject(line).points[:length.shape[0]].T
return dict(x=x, y=y, z=z)
```

Reply:
Nice! `.points` should be `.GetPoints()`, but this seems to be working fine